tests/functional/test_install_config.py

```diff
@@ -133,56 +124,55 @@ def test_command_line_appends_correctly(script, data):
     ), 'stdout: {}'.format(result.stdout)
 
 
-@pytest.mark.network
-def test_config_file_override_stack(script, virtualenv):
+def test_config_file_override_stack(
+    script, virtualenv, mock_server, shared_data
+):
     """
     Test config files (global, overriding a global config with a
     local, overriding all with a command line flag).
-
     """
-    fd, config_file = tempfile.mkstemp('-pip.cfg', 'test-')
-    try:
-        _test_config_file_override_stack(script, virtualenv, config_file)
-    finally:
-        # `os.close` is a workaround for a bug in subprocess
-        # https://bugs.python.org/issue3210
-        os.close(fd)
-        os.remove(config_file)
+    mock_server.set_responses([
+        package_page({}),
```

Comment:
I think it would help future test debugging to add a (maybe optional) `expected_path_info="/simple1/initools/"` arg to `package_page` (and other fake pages).

https://github.com/pypa/pip/blob/9ae2e6440d9fc98e91acec0a324b410fabc5452a/tests/lib/server.py#L122-L127
would return a `500 - Unexpected path_info: /foo/bar while expecting /simple1/initools/`.

Reply:
It seems like it might be easy to get lost in the possible error states - pip would've already done the wrong thing and then it will do more (unexpectedly) in response to the server error.